### PR TITLE
feat: add initial testing components for cosmos storage

### LINKED ISSUE #94: ✨ Create `TemporaryNoSqlDocument` test fixture for test-friendly Azure CosmosDb storage interaction
**Is your feature request related to a problem? Please describe.**
When interacting with Azure CosmosDb storage, we need a test-friendly way of uploading and downloading documents. Any uploading operation should automatically result in a removal of the document when the test is run.

**Describe the solution you'd like**
A good way of showing this setup/teardown relationship is by creating a temporary test fixture which uploads the document upon creation of the test fixture and removes them once the fixture is disposed of. We should make sure that we both provide factory methods for uploading a new document as well as referencing an existing document. That way, any business operation that uploads a blob is also taken into account when the test fixture is disposed.

**Additional context**
* https://www.codit.eu/blog/reusable-integration-testing-patterns-enhancing-test-configuration-test-fixtures-and-clean-slate-strategies/
* https://github.com/Codit/testing-framework/tree/main/src/Codit.Testing.CosmosDb/Helper

### PR DESCRIPTION
**Pull request by stijnmoreels**
Add initial testing components for Azure Cosmos:
* `TemporaryMongoDbCollection` & `TemporaryNoSqlContainer` - acts the same way as the `TemporaryBlobContainer`
* `TemporaryMongoDbDocument` & `TemporaryNoSqlItem` - acts the same was as the `TemporaryBlobFile`.

Closes #94

### CODE REVIEW

**Review by ClementVaillantCodit — APPROVED** [review]
Great work!! :)

File: src/Arcus.Testing.Storage.Cosmos/TemporaryNoSqlContainer.cs [hunk 1]
```diff
@@ -0,0 +1,558 @@
...
+        /// if the document was inserted by the test fixture (using <see cref="TemporaryNoSqlContainer.AddItemAsync{TItem}"/>).
+        /// </summary>
+        public OnTeardownNoSqlContainerOptions CleanCreatedItems()
+        {
+            Items = OnTeardownNoSqlContainer.CleanIfCreated;
+            return this;
+        }
+
+        /// <summary>
+        /// Configures the <see cref="TemporaryNoSqlContainer"/> to delete all the NoSql items upon disposal - even if the test fixture didn't added them.
```
Comment (stijnmoreels) [inline src/Arcus.Testing.Storage.Cosmos/TemporaryNoSqlContainer.cs:215]:
```suggestion
        /// Configures the <see cref="TemporaryNoSqlContainer"/> to delete all the NoSql items upon disposal - even if the test fixture didn't add them.
```

File: src/Arcus.Testing.Storage.Cosmos/TemporaryNoSqlContainer.cs [hunk 2]
```diff
@@ -0,0 +1,558 @@
...
+            Items = OnSetupNoSqlContainer.CleanIfExisted;
+            return this;
+        }
+
+        /// <summary>
+        /// Configures the <see cref="TemporaryNoSqlContainer"/> to delete the NoSql items
+        /// upon the test fixture creation that match any of the configured <paramref name="filters"/>.
+        /// </summary>
+        /// <remarks>
+        ///     Multiple calls will aggregated together in an OR expression.
```
Comment (stijnmoreels) [inline src/Arcus.Testing.Storage.Cosmos/TemporaryNoSqlContainer.cs:163]:
```suggestion
        ///     Multiple calls will be aggregated together in an OR expression.
```

File: src/Arcus.Testing.Core/DisposableCollection.cs [hunk 3]
```diff
@@ -170,6 +170,7 @@ await policy.ExecuteAsync(
 
             if (exceptions.Count == 1)
             {
+                _logger.LogTrace(exceptions[0].ToString());
```
Comment (stijnmoreels) [inline src/Arcus.Testing.Core/DisposableCollection.cs:173]:
```suggestion
```

File: docs/preview/02-Features/04-Storage/02-cosmos.mdx [hunk 4]
```diff
@@ -0,0 +1,231 @@
...
+```
+
+> 🎖️ The `TemporaryMongoDbCollection` will always remove any MongoDb documents that were inserted via the temporary collection itself with the `collection.InsertDocumentAsync`. This follows the 'clean environment' testing principal that any test should not leave any state it created behind after the test has run.
+
+### Temporary MongoDb document
+The `TemporaryMongoDbDocument` provides a solution when the integration test requires a document during the test run. A MongoDb document is created upon the setup of the test fixture and is deleted again when the test fixture is disposed.
+
+When the document already existed (already a document with a same configured document ID), then the test fixture will revert to the original content of the document upon the test fixture disposal.
+
+> ⚡ Whether or not the test fixture should use an existing document is set by using n existing document ID (`_id` by default).
```
Comment (stijnmoreels) [inline docs/preview/02-Features/04-Storage/02-cosmos.mdx:105]:
```suggestion
> ⚡ Whether or not the test fixture should use an existing document is set by using an existing document ID (`_id` by default).
```

**Pull request merged**